src/inmanta/ast/type.py

```diff
@@ -329,6 +450,7 @@ def validate(self, value: Optional[object]) -> bool:
         raise RuntimeException(None, f"Invalid value '{value}', expected {self.type_string()}")
 
     def cast(self, value: Optional[object]) -> object:
+        # TODO: this is a bit odd, in that is accepts None?
```

Comment:
@sanderr what do you think of this?

Reply:
I mean, it can really get any value, we don't do any validation before we call this. And I think that's good for a cast function. Whether the explicit `Optional` annotation is useful here is a different matter. I often use it, even around `object` to draw explicit attention to the fact that a value might be `None`, but in this case I'm not sure that there's value to that. I'd be fine with making it just `object`.

Reply:
Sorry, I mean the implementation  one line below

Reply:
Hmm, I agree it's strange. But I don't think we can change this on a minor. Currently `bool(null)` is simply accepted and becomes `false`. And perhaps there is something to say about it.